### ISSUE TITLE
Supporting "not within an EM" at text-combine-upright and tts:textCombine for 縦中横 'tate-chu-yoko'

### ISSUE THREAD

**Issue by akiseino** [w3c/ttml2#1127]
Can you support both of "within an EM " and "not within an EM " 
at `text-combine-upright` and `tts:textCombine` ?

I believe that 縦中横 'tate-chu-yoko' within an EM is supported by w3c / ttml2.
It can use up to 2 digits like 22.
But it's hard to read if it's 3 or 4 digits like 222 or 2222 within an EM. Because it's getting small.

Please support "not within an EM" for 3 and 4 digits to keep same size letter.
It means 2 digits letter within an EM and 3 (and 4) digits letter not within an EM should be same size.
To keep same size, it's necessary 0.25EM at both side in 3 digits and 0.5EM at both side in 4 digits like below.

![TateChuYoko_digits](https://user-images.githubusercontent.com/39544121/60810200-7da46380-a1c7-11e9-8117-2ce85263c78b.png)

At MS Word, we can select both by
Check box for "Fit in Line" at Horizontal in Vertical in Home Tab.

It would be very help if we can select both for the intent like above MS Word.
And it should be careful for 2nd line to avoid interference at the subtitling process.

Thank you.

**Comment by skynavga** [comment]
Thanks for your question Aki. The intention of the current specification of `tts:textCombine` is to permit an implementation to use a variety of techniques, which include both techniques that fit the combination into an EM square, "within an EM", as well as techniques that go outside of the EM square, "not within an EM".

In section §10.2.42 appears the following language, where I have  highlighted the last sentence that supports "not within an EM" techniques.

>Combination processing may make use of one or more techniques to obtain the goal of visual combination into an EM square of the surrounding vertical text. For example, half-width variant forms may be selected, a ligature may be selected, a smaller font size may be applied, etc. At a minimum, an implementation that supports this style property must be able to select half-width variant forms if available. **If none of these techniques are able to achieve the target dimension along the block progression dimension of the containing line area, then this dimension of the containing line area may be increased if permitted by the line stacking strategy in effect.**

Since TTML2 does not prescribe which rendering techniques are used (or favored) by an implementation, it is up to each implementation to decide whether a given combination should use an "in an EM" or a "not within an EM" rendering technique.

If you have any further questions or would suggest we modify the current text, please say so. Otherwise, if you are satisfied by this response, feel free to close this issue as you wish.

**Comment by palemieux** [comment]
>  it is up to each implementation to decide whether a given combination should use an "in an EM" or a "not within an EM" rendering technique.

It looks like @akiseino is asking that the author be able to specify a preference for not fitting characters at all within an EM.

As it stands, TTML2 and CSS clearly express a preference for _typeset[ing] all consecutive characters within the box horizontally_.

As an important data point, digital cinema subtitles behave the other way, and do not require/encourage/suggest that implementations typeset all consecutive characters within the box horizontally.

**Comment by akiseino** [comment]
Thank you for your answer, Glenn-san.

As @palemieux mentioned that, we typically use 4 digit like "2019" as tate-cyu-yoko to show year in vertical Japanese subtitle in digital cinema. And it contains some intents from director and translator for "looking good".

In that case, we prefer to use that as 4 digit sample which attached image in my previous for keeping the original intent. 

To archive that, it's necessary 2 EM width instead of using smaller letter.
I'd like to ask to have option for that if it's impossible.
Thank you.

**Comment by skynavga** [comment]
@akiseino just to be clear, are you asking for an enhancement in TTML that allows an author to specify a preference for using an "inside EM" versus "outside EM" technique in the case that an implementation supports multiple techniques? or are you satisfied with the current specification text which makes this choice implementation dependent?

**Comment by akiseino** [comment]
@skynavga I'd like to ask enhancement in TTML that allows an author to specify a preference for using an "inside EM" versus "outside EM" technique in the case that an implementation supports multiple techniques. It would be very help for Japanese content. Thank you.

**Comment by skynavga** [comment]
@akiseino thanks for that clarification; I suspect this will be handled via a ruby extensions module being prepared, which will appear under [1]; I will bring to attention of WG for further processing

[1] https://github.com/w3c/tt-module-ruby-ext-1

**Comment by skynavga** [comment]
If there is no objection, I would like to move this issue to https://github.com/w3c/tt-module-cjk-ext-1, after which I will add a comment with a link to this new location in https://github.com/w3c/tt-reqs/issues/12.

**Comment by palemieux** [comment]
- any reason not to use https://help.github.com/en/articles/transferring-an-issue-to-another-repository ?
- can the repos for the TTWG-managed modules be listed at https://github.com/w3c/ttml2/blob/master/README.md?

**Comment by skynavga** [comment]
@palemieux that is how I would move (transfer) this issue, a process I have used previously, e.g., to move TTML2 issues into the TTML3 repo; I will add pointers to the README once we have merged #1096 (for which I requested your re-review yesterday);

**Comment by palemieux** [comment]
@skynavga By the way, how would a module add a permitted value to a core style property? A modules can easily add foreign attributes/elements since they are pruned/ignored by implementations that do not understand the module. The same is not true about values of style properties.

**Comment by skynavga** [comment]
@palemieux it could not; it would have to be added by modifying the original specification of that property, which would have to be done in a new major version (since this would be a substantive functional addition), or, if it could be argued that the modification were a bug fix (e.g., a value should have been specified but was inadvertently omitted), then it could be done in a new edition (i.e., new minor version);

**Comment by palemieux** [comment]
Makes sense. So perhaps it is too soon to move this issue to another repo since a solution might be to introduce additional values for tts:textCombine.

**Comment by skynavga** [comment]
In this case, it would have to be moved to TTML3, since this is definitely not in the bug fix category. I was planning to draft this as a new property, e.g., `tts:textCombineStrategy`, which would be at home in the cjk-ext-1 module and not require modifications to any version of the TTML core modules.

**Comment by palemieux** [comment]
> tts:textCombineStrategy

This could work, but will ultimately depend on CSS WG's feedback, right? They may introduce additional values to `text-combine`.

I suggest just leaving the issue here for now. Just a suggestion...

**Comment by skynavga** [comment]
Even if CSSWG were to introduce new values to `text-combine`, that does not mean we would have to do so. A TTML renderer that uses CSS could merely map a combination of `tts:textCombine` and `tts:textCombineStrategy` to future defined CSS `text-combine` values. If we were going to add one more new values to `tts:textCombine` itself, we would have to do so in TTML3, so, in any case, this issue needs to move somewhere that is not TTML2. I'm going to go ahead and move it to https://github.com/w3c/tt-module-cjk-ext-1 where I can draft a preliminary proposal to add a `tts:textCombineStrategy` property to handle this enhancement request. Then we will have something concrete to point at in order to discuss the technical details and interface with the CSSWG.
